### PR TITLE
Add the Lcd demo app

### PR DESCRIPTION
**Pull request by engdoreis**
This PR:

- Add a spi driver
- Add the driver for SST7735S LCD as a submodule.
- Add a demo app using the LCD.

### CODE REVIEW

File: sw/CMakeLists.txt
Comment (engdoreis) [reply sw/CMakeLists.txt:7]: No, removing!

File: .gitmodules
Comment (engdoreis) [reply .gitmodules:3]: Do I have access to do so? Or maybe @GregAC could fork the repo.

File: sw/demo/lcd_st7735/main.c
Comment (engdoreis) [reply sw/demo/lcd_st7735/main.c:80]: Fixed.

File: sw/common/spi.c
Comment (engdoreis) [reply sw/common/spi.c]: Fixed

File: sw/demo/lcd_st7735/lcd.c
Comment (engdoreis) [reply sw/demo/lcd_st7735/lcd.c:5]: Added.

File: sw/demo/lcd_st7735/lcd.h
Comment (engdoreis) [reply sw/demo/lcd_st7735/lcd.h:5]: added

File: sw/demo/lcd_st7735/main.c
Comment (engdoreis) [reply sw/demo/lcd_st7735/main.c:5]: added

File: sw/demo/lcd_st7735/CMakeLists.txt
Comment (marnovandermaas) [reply sw/demo/lcd_st7735/CMakeLists.txt:1]: Yes, you're right. Should not be necessary then.

File: sw/common/demo_system.h
Comment (marnovandermaas) [inline sw/common/demo_system.h:16]: I've chosen the same address for the PWM. I guess we can wait until the first PR is merged and then the second one has to change the value to `0x80004000` or so

File: sw/common/spi.c
Comment (marnovandermaas) [inline sw/common/spi.c:12]: Nit: just based on the vendored in code base, I think the consensus is adding a space between the type name and the curly braces.

File: sw/common/spi.h
Comment (marnovandermaas) [inline sw/common/spi.h:18]: Nit: newline after struct definition?

Comment (marnovandermaas) [inline sw/common/spi.h:18]: Nit: newline after struct definition?

File: sw/demo/lcd_st7735/fractal.c
Comment (marnovandermaas) [inline sw/demo/lcd_st7735/fractal.c:137]: Nit: newline at end of file.

File: sw/demo/lcd_st7735/CMakeLists.txt
Comment (marnovandermaas) [inline sw/demo/lcd_st7735/CMakeLists.txt:16]: Nit: newline at end of file.

**Review by marnovandermaas — COMMENTED** [review]
Nice progress on the fractal and on cleaning up the LCD driver. We do need to make sure the SPI PR gets merged before we merge this one, so I would convert this to a draft in the meantime.

File: sw/CMakeLists.txt
Comment (GregAC) [inline sw/CMakeLists.txt:7]: Is there something we're using from stdlib?

File: sw/common/spi.c
Comment (GregAC) [inline sw/common/spi.c:21]: Better off to have the `while` first? Then we're guaranteed space in the transmit FIFO when we write to it

File: sw/common/spi.h
Comment (GregAC) [inline sw/common/spi.h:30]: Are these left over from when this was a bit-banging driver?

File: .gitmodules
Comment (GregAC) [reply .gitmodules:3]: Yeah I can bring it in as a lowRISC repo.

Though I'd like to vendor it in rather than use git submodules to match our approach for other things in the demo system

**Review by GregAC — COMMENTED** [review]
This is looking good, just need to switch to a vendored display driver and I'd also like to see some more comments in the code. Given this is targeted at beginners I think that will help understanding, even if the code is straight-forward to an experienced software developer.

**Comment by GregAC** [comment]
To vendor in the display driver see the OT docs here: https://docs.opentitan.org/doc/ug/vendor_hw/#how-to-vendor-new-code use the vendor.py tool from the OT repository (which we should probably add in here too but don't worry about that for now).

File: sw/CMakeLists.txt
Comment (engdoreis) [reply sw/CMakeLists.txt:7]: The sw implementation of floating point.

File: sw/common/spi.c
Comment (engdoreis) [reply sw/common/spi.c:12]: Fixed

File: sw/common/spi.h
Comment (engdoreis) [reply sw/common/spi.h:18]: Done!

File: .gitmodules
Comment (engdoreis) [reply .gitmodules:3]: Done

File: sw/demo/lcd_st7735/CMakeLists.txt
Comment (engdoreis) [reply sw/demo/lcd_st7735/CMakeLists.txt:16]: Fixed

File: sw/demo/lcd_st7735/lcd.c
Comment (engdoreis) [reply sw/demo/lcd_st7735/lcd.c:20]: Done

**Comment by engdoreis** [comment]
> To vendor in the display driver see the OT docs here: https://docs.opentitan.org/doc/ug/vendor_hw/#how-to-vendor-new-code use the vendor.py tool from the OT repository (which we should probably add in here too but don't worry about that for now).

Display driver vendored.

File: sw/demo/lcd_st7735/main.c
Comment (GregAC) [reply sw/demo/lcd_st7735/main.c:150]: Commented out code is still there

**Review by GregAC — APPROVED** [review]

**Pull request merged**
